File: src/main/java/com/bvanseg/gigeresque/client/entity/render/AquaticAlienEntityRenderer.java
package com.bvanseg.gigeresque.client.entity.render;

import com.bvanseg.gigeresque.client.entity.model.AquaticAlienEntityModel;
import com.bvanseg.gigeresque.client.entity.render.feature.AquaticAlienFeatureRenderer;
import com.bvanseg.gigeresque.common.entity.impl.AquaticAlienEntity;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.render.VertexConsumerProvider;
import net.minecraft.client.render.entity.EntityRendererFactory;
import net.minecraft.client.util.math.MatrixStack;
import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer;

@Environment(EnvType.CLIENT)
public class AquaticAlienEntityRenderer extends GeoEntityRenderer<AquaticAlienEntity> {
	public AquaticAlienEntityRenderer(EntityRendererFactory.Context context) {
		super(context, new AquaticAlienEntityModel());
		this.shadowRadius = 0.5f;
		this.addLayer(new AquaticAlienFeatureRenderer(this));
	}

	@Override
	public void render(AquaticAlienEntity entity, float entityYaw, float partialTicks, MatrixStack stack,
			VertexConsumerProvider bufferIn, int packedLightIn) {
		float scaleFactor = 0.5f + ((entity.getGrowth() / entity.getMaxGrowth()) / 5f);
		stack.scale(scaleFactor, scaleFactor, scaleFactor);
		super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn);
	}
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/AquaticAlienEntityRenderer.kt
package com.bvanseg.gigeresque.client.entity.render

import com.bvanseg.gigeresque.client.entity.model.AquaticAlienEntityModel
import com.bvanseg.gigeresque.client.entity.render.feature.AquaticAlienFeatureRenderer
import com.bvanseg.gigeresque.client.extensions.scale
import com.bvanseg.gigeresque.common.entity.impl.AquaticAlienEntity
import net.minecraft.client.render.VertexConsumerProvider
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.util.math.MatrixStack
import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer

/**
 * @author Boston Vanseghi
 */
class AquaticAlienEntityRenderer(context: EntityRendererFactory.Context) : GeoEntityRenderer<AquaticAlienEntity>(
    context,
    AquaticAlienEntityModel()
) {
    init {
        this.shadowRadius = 0.5f
        this.addLayer(AquaticAlienFeatureRenderer(this))
    }

    override fun render(
        entity: AquaticAlienEntity,
        entityYaw: Float,
        partialTicks: Float,
        stack: MatrixStack,
        bufferIn: VertexConsumerProvider,
        packedLightIn: Int
    ) {
        stack.scale(0.5f + ((entity.growth / entity.maxGrowth.toFloat()) / 5f))
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
    }
}